### PR TITLE
Port parallel storage access to 1.5.0.

### PR DESCRIPTION
**Pull request by goral09**

### CODE REVIEW

File: node/src/components/storage.rs [hunk 1]
```diff
@@ -840,12 +787,9 @@ impl Storage {
...
     fn handle_storage_request<REv>(
-        &mut self,
+        &self,
         req: StorageRequest,
-    ) -> Result<Effects<Event>, FatalStorageError>
-    where
-        Self: Component<REv>,
-    {
+    ) -> Result<Effects<Event>, FatalStorageError> {
         // Note: Database IO is handled in a blocking fashion on purpose throughout this function.
```
Comment (marc-casperlabs) [inline node/src/components/storage.rs:793]: Sidenote: This comment is outdated and should be removed from `dev` :)

**Review by marc-casperlabs — APPROVED** [review]
Looks good to me!

**Comment by goral09** [comment]
> Looks good to me!

Except that it won't work (yet) b/c we haven't ported https://github.com/casper-network/casper-node/pull/2844 to `dev`.

**Review by dwerner — APPROVED** [review]

**Comment by goral09** [comment]
Closing this PR as its changes were merged as part of https://github.com/casper-network/casper-node/pull/2857.

**Pull request closed without merging**